presto-orc/src/main/java/io/prestosql/orc/writer/TimestampColumnWriter.java

```diff
@@ -316,36 +319,17 @@ public void reset()
         statisticsBuilder = statisticsBuilderSupplier.get();
     }
 
-    private void writeTimestampMicros(Block block)
-    {
-        for (int i = 0; i < block.getPositionCount(); i++) {
-            if (!block.isNull(i)) {
-                long micros = type.getLong(block, i);
-
-                // ORC erroneously uses normal division for seconds
-                long seconds = micros / MICROSECONDS_PER_SECOND;
-                long microsFraction = floorMod(micros, MICROSECONDS_PER_SECOND);
-                long nanosFraction = microsFraction * NANOSECONDS_PER_MICROSECOND;
-
-                long millis = floorDiv(micros, MICROSECONDS_PER_MILLISECOND);
-
-                writeValues(seconds, nanosFraction);
-                statisticsBuilder.addValue(millis);
-            }
-        }
-    }
-
-    private void writeTimestampNanos(Block block)
+    private void writeTimestamp(Block block)
     {
         for (int i = 0; i < block.getPositionCount(); i++) {
             if (!block.isNull(i)) {
-                LongTimestamp timestamp = (LongTimestamp) type.getObject(block, i);
+                LongTimestamp timestamp = longTimestampRepresentation(block, i, (TimestampType) type);
 
                 // ORC erroneously uses normal division for seconds
                 long seconds = timestamp.getEpochMicros() / MICROSECONDS_PER_SECOND;
                 long microsFraction = floorMod(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
                 long nanosFraction = (microsFraction * NANOSECONDS_PER_MICROSECOND) +
-                        (timestamp.getPicosOfMicro() / PICOSECONDS_PER_NANOSECOND);
```

Comment:
@electrum this was not intentional?

Reply:
This change is not needed, since this code is specifically for `timestamp(9)` and thus there is no difference between truncation and rounding. Please remove the `roundDiv` and `if` as it's not needed.